src/Avalonia.Controls/Panel.cs

```diff
@@ -161,6 +170,12 @@ private protected virtual void InvalidateMeasureOnChildrenChanged()
             InvalidateMeasure();
         }
 
+        private void ChildrenPropertyChanged(object? sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Children.Count) || e.PropertyName is null)
+                _totalCountChanged?.Invoke(this, EventArgs.Empty);
+        }
```

Comment:
There is a lot of panels used in the application. Would it make sense to subscribe on Children.PropertyChanged only when TotalCountChanged has any subscription? Could be a minor one though.

Reply:
Fixed.